scylla-persistence/src/main/java/com/netflix/conductor/scylla/dao/ScyllaExecutionDAO.java

```diff
@@ -333,38 +347,71 @@ public List<TaskModel> createTasks(List<TaskModel> tasks) {
-                session.execute(
-                        selectTaskInProgressStatement.bind(task.getTaskDefName(),
-                                UUID.fromString(task.getTaskId())));
-        if (resultSet.all().isEmpty() || resultSet.all().size()<1) {
-            session.execute(
-                    insertTaskInProgressStatement.bind(task.getTaskDefName(),
-                            UUID.fromString(task.getTaskId()),
-                            UUID.fromString(task.getWorkflowInstanceId()),
-                            true));
+        ResultSet resultSet = getTaskInProgressTaskFromTaskModel(task);
+        if (resultSet.all().size() < 1) {
+            session.execute(insertTaskInProgressV2Statement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId()),
+                    UUID.fromString(task.getWorkflowInstanceId()), true));
+        } else {
+            LOGGER.info("Task with defName {} and Id {} and status {} in addTaskInProgress NOT inserted as already exists  ", task.getTaskDefName(),
+                    task.getTaskId(), task.getStatus());
         }
-        else {
-            LOGGER.info("Task with defName {} and Id {} and status {} in addTaskInProgress NOT inserted as already exists  "
-                    ,task.getTaskDefName(), task.getTaskId(),task.getStatus());
+    }
+
+    private ResultSet getTaskInProgressTaskFromTaskModel(TaskModel task) {
+        try {
+            ResultSet resultSet = fetchFromTaskInProgressV2Query(task);
+            if (isResultSetNotValid(resultSet)) {
+                return fetchFromTaskInProgressQuery(task);
+            }
+            return resultSet;
+        } catch (Exception ex) {
+            LOGGER.error("Error while reading task from task_in_progress_v2 table - taskDefName: {}, taskId: {}, status: {}",
+                    task.getTaskDefName(), task.getTaskId(), task.getStatus(), ex);
+            return null;
         }
+    }
+
+    private ResultSet fetchFromTaskInProgressV2Query(TaskModel task) {
+        return session.execute(selectTaskInProgressV2Statement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId())));
+    }
+
+    private ResultSet fetchFromTaskInProgressQuery(TaskModel task) {
+        logTaskInProgressStatus(task);
+        return session.execute(selectTaskInProgressStatement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId())));
+    }
 
+    private boolean isResultSetNotValid(ResultSet resultSet) {
+        return Objects.isNull(resultSet) || resultSet.all().size() < 1;
     }
 
     /**
      * @method to remove the task_in_progress table with the status of the task
      */
     public void removeTaskInProgress(TaskModel task) {
-        session.execute(
-                deleteTaskInProgressStatement.bind(task.getTaskDefName(),UUID.fromString(task.getTaskId())));
+        ResultSet resultSet = session.execute(
+                deleteTaskInProgressV2Statement.bind(task.getTaskDefName(), UUID.fromString(task.getTaskId())));
```

Comment:
Have we verified what happens if the row itself is not present and we attempt to delete it? Specifically, does the result.wasApplied set return true or false in such a scenario?
If Yes then we might need to run 2 queries.

Reply:
`resultSet.wasApplied()` is wrong here, putting both in batch, similar for other